### PR TITLE
Use cats-mtl to derive LambdaEnv instances

### PR DESCRIPTION
**Pull request by kubukoz**

### CODE REVIEW

File: lambda/shared/src/test/scala/feral/lambda/LambdaEnvLiftSuite.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  implicit val env: LambdaEnv[IO, Event]
+
+  implicitly[LambdaEnv[Kleisli[IO, String, *], Event]]
+  implicitly[LambdaEnv[WriterT[IO, String, *], Event]]
+  implicitly[LambdaEnv[StateT[IO, String, *], Event]]
+  implicitly[LambdaEnv[EitherT[IO, String, *], Event]]
+  implicitly[LambdaEnv[OptionT[IO, *], Event]]
+  // doesn't work with this PR, works without it
+  // implicitly[LambdaEnv[Kleisli[EitherT[IO, String, *], String, *], Event]]
```
Comment (armanbilge) [inline lambda/shared/src/test/scala/feral/lambda/LambdaEnvLiftSuite.scala:38]: Update: this works on Scala 3 with this PR.

Comment (armanbilge) [reply lambda/shared/src/test/scala/feral/lambda/LambdaEnvLiftSuite.scala:38]: Update: nevermind, can't read 😆

Comment (kubukoz) [reply lambda/shared/src/test/scala/feral/lambda/LambdaEnvLiftSuite.scala:38]: I miscommunicated, it doesn't :D

**Comment by armanbilge** [comment]
This has bincompat implications so we have to decide before 0.1.0 :)

**Comment by kubukoz** [comment]
tbh I think it was better before this, at least we could support arbitrary compositions of transformers. This way we only get one level... which is less than ideal.

**Comment by armanbilge** [comment]
It's annoying boilerplate but not the end of the world.

**Pull request closed without merging**